llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -401,5 +401,10 @@ bool RISCVTTIImpl::isSourceOfDivergence(const Value *V) const {
   if (isa<InvokeInst>(V))
     return true;
 
+  if (dyn_cast<PHINode>(V) && 
+      dyn_cast<Instruction>(V)->getParent()->getParent()->getCallingConv() !=
+      CallingConv::VENTUS_KERNEL) 
```

Comment:
I think all the instructions in a non kernel function should be considered as divergency instructions.

Because you don't know if a function is invoked in a divergency path or or uniformed path.